src/main/java/io/stargate/sgv2/jsonapi/service/resolver/sort/TableCqlSortClauseResolver.java

```diff
@@ -142,6 +149,48 @@ private WithWarnings<OrderByCqlClause> resolveNonVectorSort(
       return WithWarnings.of(OrderByCqlClause.NO_OP, warn);
     }
 
+    // We know all the order keys are partition sorting keys and no clustering key missing or
+    // skipping.
+    // If not all partitionKeys are restricted by $eq, enable in memory sort
+    // CQL has below rules:
+    // 1. "ORDER BY is only supported when the partition key is
+    // restricted by an EQ or an IN."
+    // 2. "Cannot page queries with both ORDER BY and a IN restriction on the partition key; you
+    // must either remove the ORDER BY or the IN and sort client side, or disable paging for this
+    // query"
+    Set<CqlIdentifier> validPartitionKey = new HashSet<>();
```

Comment:
as discussed, look at adding a proporty to the WhereCQLClause that returns true if the where clause has the necessary restirctions, and move this logic into the building of the TableWhereCqlClause